### PR TITLE
[Work in progress] 8372 sync and async zio-s share the same task queues

### PR DESCRIPTION
**Pull request by avg-I**

### CODE REVIEW

**Comment by avg-I** [comment]
In addition to dispatching synchronous zio-s via "high" task queues, I also modified the configuration of those task queues to match the regular counterparts. I did that to avoid any performance regressions.
But it could be that the new values are an overkill.  Especially, for the synchronous writes of which there not should be that many.  At least, for now while we have a single thread issuing all ZIL writes.

So, any feedback and suggestions on that adjustment is very appreciated.

**Comment by prakashsurya** [comment]
@avg-I can you rebase this onto the latest `master` branch? it's failing the automated build/test because there's changes to the `jenkins` directory in `master` that don't exist in your branch/PR (i.e. commits that exist on `master` that aren't in your PR branch).

**Comment by avg-I** [comment]
Oh, I wondered about that strange failure...
Rebased now.

File: usr/src/uts/common/fs/zfs/zio.c [hunk 1]
```diff
@@ -1417,6 +1417,13 @@ zio_taskq_dispatch(zio_t *zio, zio_taskq_type_t q, boolean_t cutinline)
 	spa_t *spa = zio->io_spa;
 	zio_type_t t = zio->io_type;
 	int flags = (cutinline ? TQ_FRONT : 0);
+	zio_priority_t p = zio->io_priority;
+
+	ASSERT(q == ZIO_TASKQ_ISSUE || q == ZIO_TASKQ_INTERRUPT);
+	if (p == ZIO_PRIORITY_SYNC_WRITE)
+		ASSERT3U(t, ==, ZIO_TYPE_WRITE);
```
Comment (ahrens) [inline usr/src/uts/common/fs/zfs/zio.c]: FYI, another way to do this is: `IMPLY(p == SYNC_WRITE, t == TYPE_WRITE)`

Comment (avg-I) [reply usr/src/uts/common/fs/zfs/zio.c]: Than you for the suggestion.  I decided to modify the code to use the more compact form.

**Comment by avg-I** [comment]
I am getting some reports that this change actually hurts performance rather than improves it.
Perhaps this is hardware dependent, perhaps not.
So, please do not commit this just yet.


I wonder if anyone else tested this change and can share any results.

**Comment by prakashsurya** [comment]
@avg-I how do you want to proceed with this change? should we move forward with the RTI? or close it if introduces a regression?

**Comment by avg-I** [comment]
@prakashsurya I am all in doubts. On the one hand, I believe that the change is correct (or, at least, is a move in the correct direction). On the other hand, I have got some contradictory feedback.

It would be great to get more performance testing.
Might Delphix be interested in that? :-)

**Comment by avg-I** [comment]
By the way, I see that 9993 zil writes can get delayed in zio pipeline (commit 2258ad0b75) has already made a step in the direction of this change with respect to ZIO_PRIORITY_SYNC_WRITE.
It hasn't updated the "high write" taskq counts, though. Not sure if that can cause a new bottleneck for ZIL writes. @grwilson, what do you think?

**Comment by ahrens** [comment]
This repo will be archived in the coming week, and therefore this PR is being closed.  Alternate processes for contributing ZFS changes (including those in open PR's) include following the [illumos procedures](https://illumos.org/docs/contributing/#code-review), or opening a PR against [ZFSonLinux](https://github.com/zfsonlinux/zfs).

The reasoning behind this are outlined in an [email](https://openzfs.topicbox.com/groups/developer/T1ea5f3fe16a66af6/openzfs-github-repo-eol) to the developer@open-zfs.org mailing list, which is reproduced below:

> The OpenZFS code repository on github (http://github.com/openzfs/openzfs) is a clone of the illumos repo, with basically identical code.  The OpenZFS repo made it easier to contribute ZFS code to illumos, by leveraging the github pull request and code review processes, and by automatically building illumos and running the ZFS Test Suite.
>
> Unfortunately, the automated systems have atrophied and we lack the effort and interest to maintain it.  Meanwhile, the illumos code review and contribution process has been working well for a lot of ZFS changes (notably including ports from Linux). 
>
>Since the utility of this repo has decreased, and the volunteer workforce isn't available to maintain it, we will be archiving http://github.com/openzfs/openzfs in the coming week.  Thank you to everyone who helped maintain this infrastructure, and to those who leveraged it to contribute over 500 commits to ZFS on illumos!  Alternate processes for contributing ZFS changes (including those in open PR's) include following the [illumos procedures](https://illumos.org/docs/contributing/#code-review), or opening a PR against [ZFSonLinux](https://github.com/zfsonlinux/zfs). 
>
>The OpenZFS github organization (http://github.com/openzfs) will continue to exist.  I will make an announcement about its future role at the [OpenZFS Developer Summit](http://open-zfs.org/wiki/OpenZFS_Developer_Summit_2019) on Monday.
>
>Let me know if you have any questions,
>--matt

**Pull request closed without merging**